File: Easy/Find duplicates in an array/find-duplicates-in-an-array.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;
import java.util.Map.Entry;

class GFG {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            int n = sc.nextInt();
            int[] a = new int[n];
            for (int i = 0; i < n; i++) a[i] = sc.nextInt();
            Solution g = new Solution();
            ArrayList<Integer> ans = g.duplicates(a, n);
            for (Integer val : ans) System.out.print(val + " ");
            System.out.println();
        }
    }
}

// } Driver Code Ends


class Solution {
    public static ArrayList<Integer> duplicates(int arr[], int n) {
        // code here
		
		Arrays.sort(arr);
	    ArrayList<Integer> list = new ArrayList<>();
	    int d = Integer.MAX_VALUE;
	    for(int i=0; i < arr.length-1; i++)
	    {
	        if(arr[i] == arr[i+1] && arr[i] != d)
	        {
	            list.add(arr[i]);
	            d = arr[i];
	        }
	    }
	    if(list.isEmpty())
	    {
	        list.add(-1);
	    }
	    return list;
    }
}

File: Easy/Find duplicates in an array/find-duplicates-in-an-array.py
class Solution:
    def duplicates(self, arr, n): 
    	
    	dic = {}
        ans = []
        for i in range(n):
            dic[i]=0
        
        for i in range(n):
            dic[arr[i]]+=1
        
        for key in range(n):
            if dic[key]>1:
                ans.append(key)
        
        return ans if ans else [-1]


#{ 
 # Driver Code Starts
if(__name__=='__main__'):
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        res = Solution().duplicates(arr, n)
        for i in res:
            print(i,end=" ")
        print()



# } Driver Code Ends